easybuild/toolchains/compiler/fujitsu.py

```diff
@@ -0,0 +1,81 @@
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Support for the Fujitsu compiler drivers (aka fcc, frt).
+
+The basic concept the same as for the Cray Programming Environment.
+
+:author: Miguel Dias Costa (National University of Singapore)
+"""
+import easybuild.tools.systemtools as systemtools
+from easybuild.tools.toolchain.compiler import Compiler, DEFAULT_OPT_LEVEL
+
+TC_CONSTANT_FUJITSU = 'Fujitsu'
+
+
+class FujitsuCompiler(Compiler):
+    """Generic support for using Fujitsu compiler drivers."""
+    TOOLCHAIN_FAMILY = TC_CONSTANT_FUJITSU
+
+    # compiler module name is lang (with version e.g. tcsds-1.2.31)
+    COMPILER_MODULE_NAME = ['lang']
+    COMPILER_FAMILY = TC_CONSTANT_FUJITSU
+
+    COMPILER_CC = 'fcc'
+    COMPILER_CXX = 'FCC'
+    COMPILER_F77 = 'frt'
+    COMPILER_F90 = 'frt'
+    COMPILER_FC = 'frt'
+
+    COMPILER_UNIQUE_OPTION_MAP = {
+        DEFAULT_OPT_LEVEL: 'Kfast',
+        'optarch': '',  # Fujitsu compiler by default generates code for the arch it is running on
+        'openmp': 'Kopenmp',
+        'unroll': 'funroll-loops',
+        'strict': ['Kfp_precision'],
+        'precise': ['Kfp_precision'],
+        'defaultprec': [],
+        'loose': ['Kfp_relaxed'],
+        'veryloose': ['Kfp_relaxed'],
+        'vectorize': {False: 'KNOSVE', True: 'KSVE'},
+    }
+
+    # used when 'optarch' toolchain option is enabled (and --optarch is not specified)
+    COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
+        # -march=archi[+features]. At least on Fugaku, these are set by default (-march=armv8.3-a+sve and -mcpu=a64fx)
+        (systemtools.AARCH64, systemtools.ARM): '',
+    }
+
+    # used with --optarch=GENERIC
+    COMPILER_GENERIC_OPTION = {
+        (systemtools.AARCH64, systemtools.ARM): '-mcpu=generic -mtune=generic',
+    }
+
+    def _set_compiler_vars(self):
+        super(FujitsuCompiler, self)._set_compiler_vars()
+
+        # enable clang compatibility mode
+        self.variables.nappend('CFLAGS', ['Nclang'])
+        self.variables.nappend('CXXFLAGS', ['Nclang'])
```

Comment:
turns out having `-Nclang` only in these flags messes up with configure scripts that probe for the compiler without them, and so detect the traditional mode instead (hitting this with the `libffi` included in `_ctypes` in `Python 2.7.18`)

one solution that seems to work is to move them to `COMPILER_CC` and `COMPILER_CCX` above (but I think then it's crucial to always enclose `$CC` etc. in double quotes, since they will include a space?)